app/assets/javascripts/discourse/app/components/discourse-topic.js

```diff
@@ -113,6 +113,7 @@ export default Component.extend(Scrolling, MobileScrollDirection, {
       ".cooked a, a.track-link",
       (e) => ClickTrack.trackClick(e, getOwner(this))
     );
+    this.header.topic = this.topic;
```

Comment:
Does this work correctly even when navigating directly between topics? (in that case, I don't think the DiscourseTopic component will be destroyed/recreated, so didInsertElement won't fire again)

Reply:
Perhaps the route afterModel() hook would be a better spot? 🤔

Reply:
Updated in https://github.com/discourse/discourse/pull/27138/commits/2cf497148d73c2581d2fe7dd6672b3b3c8b288a1